### DISCUSSION TITLE
Remove TypeDeclarationStatement without altering lines position

### DISCUSSION THREAD

**Discussion by uniquadev** [GGG-KILLER/Loretta#113] [Q&A] [answered]
Hello, I'm trying to remove `TypeDeclarationStatement`  with the `LuaSyntaxRewriter` class without altering lines..

This is how I remove TypeDeclarationStatements
```c#
public override SyntaxNode VisitTypeDeclarationStatement(TypeDeclarationStatementSyntax node)
{
    return null;
}
```
es: 
```lua
function a()
    print('epic')
end


type Point = { x: number, y: number }


function foo(x: number, y: string): boolean
    local k: string = y:rep(x)
    return k == "a"
end
```
out:
```lua
function a()
    print('epic')
end
function foo(x: number, y: string): boolean
    local k: string = y:rep(x)
    return k == "a"
end
```

Have a good day!

**Comment by GGG-KILLER** [comment]
The lines are being "altered" because the spacing in the type declaration statement was in the `LeadingTrivia` of the `type` keyword, along with probably having `.NormalizeWhitespace()` being called which removes extra line breaks when it finds them.

If you could share the rest of the process we could confirm whether or not that's the case @uniquadev

**Reply by uniquadev** [reply]
```c#
internal class LuauRewriter : LuaSyntaxRewriter
    {
        public static SyntaxNode Rewrite(SyntaxNode node)
        {
            var rewriter = new LuauRewriter();
            return rewriter.Visit(node);
        }

        // TODO make a way to don't alter lines position
        // an idea could be visit statements list and remove types declaration modifing near statements trivia
        public override SyntaxNode VisitTypeDeclarationStatement(TypeDeclarationStatementSyntax node)
        {
            return null;
        }
    }
```

**Reply by GGG-KILLER** [reply]
Well, this only seems like it would only remove the line breaks up to the type declaration and the one immediately after it:
![image](https://user-images.githubusercontent.com/5892127/210275385-51928643-97e6-4b38-b5a6-e9fe2c65d587.png)

The other line breaks should've remained if `.NormalizeWhitespace()` is not being called.

So the final code should've been
```lua
function a()
    print('epic')
end


function foo(x: number, y: string): boolean
    local k: string = y:rep(x)
    return k == "a"
end
```
if only this visitor was being ran

**Reply by uniquadev** [reply]
sorry I forgot to mention that I had another visitor. I isoleted the `LuauRewriter` class  and I got the expected result :DD
However line position is still different (even if better than the old case)

**Reply by GGG-KILLER** [reply, marked as answer]
Yeah, the way to solve this would be to take the leading trivia of the `type` keyword and add it to the `function` in `function foo`
